src/_pytest/debugging.py

```diff
@@ -154,6 +161,9 @@ def setup(self, f, tb):
     @classmethod
     def set_trace(cls, *args, **kwargs):
         """Invoke debugging via ``Pdb.set_trace``, dropping any IO capturing."""
+        if pytestPDB._config:  # Might not be available when called directly.
+            if pytestPDB._config.getoption("pdb_ignore_set_trace"):
```

Comment:
Reading the option name, I would've expected this option to only override `pdb.set_trace()`.

However this will cause it to also skip `pytest.set_trace()`, which I'm not sure is an intended behavior.

Reply:
Ah, true.

Are you using `pytest.set_trace()`?

Do you think it makes sense to not skip it?
Might be helpful to still insert a `set_trace` when using the option after all.

Reply:
I just poked at the docs for `pytest.set_trace` and it states:

`This is no longer needed and one can use the native import pdb;pdb.set_trace() call directly.`.

This is stating that the behavior between the two should be the same, so I think it actually makes sense here to have them behave the same way.

Reply:
The docs could be adjusted after all.

I am not sure myself, but cnsidered it a good hack to being able to still `__import__("pytest").set_trace()` if needed.

Anyway, I would also be happy to not having to change docs of course.. :)

Reply:
While I can think of some fringe cases where you might want to set a breakpoint when using this option, I think it would be preferred to have this option skip setting a breakpoint whenever possible.

As a side-note, I believe that `breakpoint()` uses `pdb.set_trace()` by default, so this will also disable `breakpoint()` (I don't have Python 3.7 installed currently to test).

Reply:
IMHO it should work the same for both `pytest.set_trace()` as well as `pdb.set_trace()`: `pytest.set_trace()` was the first implementation, afterwards it was improved to work on `pdb.set_trace()`, so it makes sense for both to do the same thing.